bin/notifications.sh

```diff
@@ -500,9 +500,10 @@ do
 				then
 					log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "Found notification types ${NOTIFY_FOR_PHASE_WITH_STATE[${ordered_phase_with_state}]} for ${ordered_phase_with_state}."
 					notification "${ordered_phase_with_state}" "${NOTIFY_FOR_PHASE_WITH_STATE[${ordered_phase_with_state}]}" "${_lfs_root_dir}"
+				else
```

Comment:
This change is not correct. If `${ordered_phase_with_state} `is defined in the `NOTIFICATION_ORDER_PHASE_WITH_STATE` **_array_**, then there must be a corresponding value for that `${ordered_phase_with_state}` in the `NOTIFY_FOR_PHASE_WITH_STATE` **_hash_**. If `"${NOTIFY_FOR_PHASE_WITH_STATE[${ordered_phase_with_state}]:-}"` is empty that is a configuration error and the script should always complain. Whether `"${selectedPhaseState}" == 'all'` or whether you manually specified only a specific `"${selectedPhaseState}" == "${ordered_phase_with_state}"` does not matter.